Integrations/src/main/java/io/deephaven/integrations/python/PythonDeephavenSession.java

```diff
@@ -255,7 +257,10 @@ protected Changes createDiff(PythonSnapshot from, PythonSnapshot to, RuntimeExce
 
     @Override
     protected Set<String> getVariableNames(Predicate<String> allowName) {
-        return PyLib.ensureGil(() -> scope.getKeys().filter(allowName).collect(Collectors.toUnmodifiableSet()));
+        return scope.currentScope().copy().keySet().stream()
```

Comment:
`PyDictWrappers` are `AutoCloseable`, and it would be better to close the copies we make of `currentScope()` and `mainGlobals`.